File: src/main/java/torch_java/api/nn/Module.java
package torch_java.api.nn;

import org.bytedeco.javacpp.*;
import org.bytedeco.javacpp.annotation.*;

@Platform(
        include = {
                "torch/all.h",
                "torch/nn/module.h"
        })

@Namespace("torch::nn") @NoOffset public class Module extends Pointer {

    static {
        String workingDir = System.getProperty("user.dir");
        System.load( workingDir + "/sosca/libjava_torch_lib.so");
    }

    public Module() {
        super((Pointer) null);
        allocate();
    }

    public native void allocate();
}

File: src/main/scala/torch_scala/api/nn/Module.scala
package torch_scala.api.nn

import org.bytedeco.javacpp._
import org.bytedeco.javacpp.annotation._


@Platform(include = Array(
  "torch/all.h",
  "torch/nn/module.h"
))
@Namespace("torch::nn") @NoOffset class Module() extends Pointer {


  val workingDir = System.getProperty("user.dir")
  System.load(workingDir + "/src/native/libjava_torch_lib0.so")


  allocate()

  @native def allocate(): Unit
}